src/go/k8s/internal/controller/redpanda/cluster_controller.go

```diff
@@ -135,6 +135,37 @@ func (r *ClusterReconciler) Reconcile(
 		return ctrl.Result{}, fmt.Errorf("unable to retrieve Cluster resource: %w", err)
 	}
 
+	// After every reconciliation, update status:
+	// - Set observedGeneration. The reconciler finished, everything action
+	//   performed in this run - including updating status, has been done, and has
+	//   observed this generation.
+	// - Set ClusterQuiescent condition, based on our best knowledge if there is
+	//   any outstanding work to do for the controller.
+	defer func(cluster *vectorizedv1alpha1.Cluster) {
+		updateStatus(ctx, r.Client, &vectorizedCluster, func(cluster *vectorizedv1alpha1.Cluster) {
+			// Set quiescent
+			cond := getQuiescentCondition(cluster)
+
+			// Even if no "business logic" shows us that the reconciler still waits
+			// for outstanding work, we'll treat an error returned always as "there
+			// is work to do".
+			if cond.Status == corev1.ConditionTrue && err != nil {
+				cond.Status = corev1.ConditionFalse
+				cond.Reason = "ReconcilerError"
+				cond.Message = err.Error()
```

Comment:
I would actually recommend not updating the status in the case of an unhandled error. The error should be present in the logs and the presence of an error would indicate that the generation _was not_ properly observed.